File: dispatch/java/adamma/c4dhi/claid/Module/Scheduling/ConsumerRunnable.java
package adamma.c4dhi.claid.Module.Scheduling;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.function.Consumer;

public class ConsumerRunnable<T> extends ScheduledRunnable
{
    private T data;
    private Consumer<T> consumer;

    public ConsumerRunnable(T data, Consumer<T> consumer)
    {
        super(ScheduleOnce.now());
        this.consumer = consumer;
        this.data = data;
    }

    public ConsumerRunnable(T data, Consumer<T> consumer, ScheduleDescription schedule)
    {
        super(schedule);
        this.consumer = consumer;
        this.data = data;
    }

    @Override
    public void run() {
        consumer.accept(data);
    }    
}

File: dispatch/ios/CLAID/Sources/CLAID/Module/Scheduling/ConsumerRunnable.swift
//
//  ConsumerRunnable.swift
//  CLAID
//
//  Created by Patrick Langer on 21.02.2025.
//


import Foundation

public final actor ConsumerRunnable<T>: ScheduledRunnable {
    private let data: T
    private let consumer: (T) -> Void
    private var valid: Bool = true

    public let schedule: ScheduleDescription

    
    public func isValid() async -> Bool {
        return valid
    }
    

    init(data: T, consumer: @escaping (T) -> Void) {
        self.schedule = ScheduleOnce.now()
        self.data = data
        self.consumer = consumer
    }

    init(data: T, consumer: @escaping (T) -> Void, schedule: ScheduleDescription) {
        self.schedule = schedule
        self.data = data
        self.consumer = consumer
    }

    public func run() async {
        guard await isValid() else { return }
        await consumer(data)
    }
    
    public func invalidate() async {
        valid = false
    }
}
